crates/flux-fhir-analysis/src/lib.rs

```diff
@@ -194,11 +195,30 @@ fn assoc_refinements_of(
     Ok(rty::AssocRefinements { items: predicates })
 }
 
-fn assoc_refinement_def(
+fn default_assoc_refinement_def(
+    genv: GlobalEnv,
+    trait_id: LocalDefId,
```

Comment:
I recently introduced [`MaybeExternId`](https://github.com/flux-rs/flux/blob/274edb4f110ee72640b896cdc05d6bb67bab7e45/crates/flux-middle/src/lib.rs#L417) to express in the type system that a local id may be for an extern item, it's not a perfect abstraction, but you at least have a reminder to be careful. Not all the code has been transitioned to use it but going forward we should start all the functions in this file implementing a query with a call to `genv.maybe_extern_id` and use the result to shadow the id passed as argument.